src/test/txvalidationcache_tests.cpp

```diff
@@ -156,7 +153,14 @@ BOOST_FIXTURE_TEST_CASE(checkinputs_test, TestChain100Setup)
     CScript p2sh_scriptPubKey = GetScriptForDestination(CScriptID(p2pkh_scriptPubKey));
     CScript p2wpkh_scriptPubKey = GetScriptForWitness(p2pkh_scriptPubKey);
 
-    const CTransactionRef p2pkh_coinbase = CreateAndProcessBlock({}, p2pkh_scriptPubKey).vtx[0];
+    bool processed;
+    const CTransactionRef p2pkh_coinbase = CreateAndProcessBlock({}, p2pkh_scriptPubKey, &processed).vtx[0];
+    BOOST_CHECK(processed);
+
+    // Make the reward with p2pkh_scriptPubKey mature
+    for (int i = 0; i < COINBASE_MATURITY + 1; ++i) {
+      CreateAndProcessBlock({}, GetScriptForDestination(WitnessV0KeyHash(coinbaseKey.GetPubKey().GetID())));
```

Comment:
nit: should we check that such blocks were processed? Like we do with the block above

Reply:
Well we always can, I think is superfluous though. The block above you mention is important since we use the coinbase to verify things in the tests, the following blocks are just made to make it mature, this is not necessarily the place where I need to that that. There are already places testing for coinbase maturity elsewhere.

Reply:
`CreatAndProcessBlock` checks that the block is processed when `processed` is not passed to it.
```
assert(processed != nullptr || was_processed);
```